didcomm/message.py

```diff
@@ -21,9 +51,13 @@ class GenericMessage(Generic[T]):
     - `pack_plaintext` to build a Plaintext DIDComm message
     """
 
-    id: str
     type: str
     body: T
+    id: Optional[Union[str, Callable]] = attr.ib(
```

Comment:
`id` is a required field in any Message. I would not make it optional, and not add any converters here.
I would let the user to explicitly set it when working with a Message.
In case of Forward message, we can have some default generators (and option to pass a custom one).

Reply:
>id is a required field in any Message

the current logic make it optional only at object initialization time but the initialization itself ensures it is presented (converter will do that): if not passed, generated

> I would let the user to explicitly set it when working with a Message.

the only requirement in spec here is to  make it unique to the sender so why do you think we need to force user here?

Reply:
updated